frontend/src/profile/Profile.tsx

```diff
@@ -0,0 +1,440 @@
+            {!props.isEdit && 
+                <ItemEntry> {val} </ItemEntry>
+            }
+        </ProfileEntryContainer>
+    );
+}
+
+const ProfileCoop = (props: ProfileCoopProps) => {
+    const val = props.coop != "" ? props.coop : "None Selected";
+    return (
+        <ProfileEntryContainer>
+            <ItemTitle> Co-op Cycle </ItemTitle>
+            {props.isEdit && 
+                <Autocomplete
+                    disableListWrap
+                    options={props.plans[props.major.name].map((p: Schedule) =>
+                        planToString(p)
+                    )}
+                    renderInput={params => (
+                    <TextField
+                        {...params}
+                        variant="outlined"
+                        fullWidth
+                    />
+                    )}
+                    value={val}
+                    onChange={(event: React.SyntheticEvent<{}>, value: any) => 
+                        props.setCoop(value)}
+                />
+            }
+            {!props.isEdit && 
+                <ItemEntry> {val} </ItemEntry>
+            }
+        </ProfileEntryContainer>
+    );
+}
+
+/* 
+TODO: // Add Advsisors to profile page once we support them
+const ProfileAdvisor = (props: any) => {
+    return (
+        <ProfileEntryContainer>
+            <ItemTitle> Advisor </ItemTitle>
+            {props.isEdit && 
+                <Autocomplete
+                    disableListWrap
+                    options={["advisor", "advisor 2", "advisor 3"]}
+                    renderInput={params => (
+                    <TextField
+                        {...params}
+                        variant="outlined"
+                        fullWidth
+                    />
+                    )}
+                    value={props.advisor}
+                    onChange={(event: React.SyntheticEvent<{}>, value: any) => props.setAdvisor(value)}
+                />
+            }
+            {!props.isEdit && 
+                <ItemEntry> {props.advisor} </ItemEntry>
+            }
+        </ProfileEntryContainer>
+    );
+}
+*/
+
+const ProfileEmail = (props: ProfileEmailProps) => {
+    return (
+        <ProfileEntryContainer>
+            <ItemTitle> Email </ItemTitle>
+            {props.isEdit && 
+                <TextField
+                    id="outlined-basic"
+                    variant="outlined"
+                    value={props.email}
+                    onChange={e => props.setEmail(e.target.value)}
+                    placeholder="johnsmith@email.com"
+                />
+            }
+            {!props.isEdit && 
+                <ItemEntry> {props.email} </ItemEntry>
+            }
+        </ProfileEntryContainer>
+    );
+}
+
+const save = (props: SaveProps) => {
+    props.setEdit(false);
+    props.setMajorAction(props.major);
+    props.setFullNameAction(props.name);
+    props.setEmailAction(props.email);
+    props.setCoopCycleAction(props.coop);
```

Comment:
setCoopCycleAction takes a Schedule, so props.coop needs to be converted to a Schedule before setCoopCycleAction is called (a similar thing is done in the OnboardingInfoScreen for reference)

Reply:
This is actually setUserCoopCycleAction, I can rename it to make it clearer. The userCoopCycleAction just takes in a string.